kylin-spark-project/kylin-spark-engine/src/main/scala/org/apache/kylin/engine/spark/builder/CubeSnapshotBuilder.scala

```diff
@@ -183,6 +185,22 @@ class CubeSnapshotBuilder extends Logging {
   }
   import org.apache.kylin.engine.spark.utils.SparkDataSource._
 
+  def checkDupKey() = {
+    val joinDescs = seg.joindescs
+    joinDescs.foreach {
+      joinDesc =>
+        val tableInfo = joinDesc.lookupTable
+        val lookupTableName = tableInfo.tableName
+        val df = ss.table(tableInfo)
+        val countColumn = df.count()
+        val lookupTablePKS = joinDesc.PKS.map(lookupTablePK => lookupTablePK.columnName)
+        val countDistinctColumn = df.agg(countDistinct(joinDesc.PKS(0).columnName, lookupTablePKS: _*)).collect().map(_(0)).toList(0)
```

Comment:
better change to : `df.agg(countDistinct(lookupTablePKS(0), lookupTablePKS.tail : _*)).collect().map(_.getLong(0)).head`

Reply:
Thank you for your review @xiaowangyu @zzcclp  I have updated this PR.